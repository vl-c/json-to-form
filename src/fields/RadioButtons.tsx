import * as React from 'react';

export interface RadioButtonsProps {
  id: string;
  label?: string;
  items: {
    value: string;
    label: string;
    checked?: boolean;
    disabled?: boolean;
  }[];
}

const RadioButtons = ({ id, label, items }: RadioButtonsProps): JSX.Element => (
  <div className="input-block">
    <span className="input-label">{label}</span>
    <div className="input-radio">
      {items &&
        items.map(({ value, label, checked, disabled }) => (
          <label key={value}>
            <input
              type="radio"
              name={id}
              value={value}
              disabled={Boolean(disabled)}
              defaultChecked={Boolean(checked)}
            />
            {label}
          </label>
        ))}
    </div>
  </div>
);

export default RadioButtons;
