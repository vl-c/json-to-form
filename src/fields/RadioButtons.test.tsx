import * as React from 'react';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import RadioButtons from './RadioButtons';

const items = [
  {
    value: 'mail',
    label: 'mail',
  },
  {
    value: 'phone',
    label: 'phone',
  },
];

function renderRadioButtons() {
  return shallow(
    <RadioButtons id={'test'} label={'RadioButtons'} items={items} />
  );
}

it('RadioButtons shallow render', () => {
  const res = renderRadioButtons();
  expect(shallowToJson(res)).toMatchSnapshot();
});
