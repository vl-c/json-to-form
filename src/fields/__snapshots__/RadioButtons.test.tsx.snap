// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons shallow render 1`] = `
<div
  className="input-block"
>
  <span
    className="input-label"
  >
    RadioButtons
  </span>
  <div
    className="input-radio"
  >
    <label
      key="mail"
    >
      <input
        defaultChecked={false}
        disabled={false}
        name="test"
        type="radio"
        value="mail"
      />
      mail
    </label>
    <label
      key="phone"
    >
      <input
        defaultChecked={false}
        disabled={false}
        name="test"
        type="radio"
        value="phone"
      />
      phone
    </label>
  </div>
</div>
`;
